spinn_utilities/timer.py

```diff
@@ -12,8 +12,42 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import sys
 
-import datetime
+
+if sys.version_info >= (3, 7):
+    # acquire the most accurate measurement available (monotonic_ns)
+    from time import monotonic_ns as now # pylint: disable=no-name-in-module
+
+    # have to convert to a timedelta for rest of code to read
+    from future.backports.datetime import timedelta # pylint: disable=no-name-in-module
+
+    # conversion factor
+    NANO_TO_MICRO = 1000.0
+
+    # as montonic_ns is nano seconds, and time delta lowest is micro,
+    # need to convert
+    def convert_to_timedelta(time_diff):
+        return timedelta(microseconds=time_diff / NANO_TO_MICRO)
+elif sys.version_info >= (3, 3):
+    # acquire the most accurate measurement available (monotonic)
+    from time import monotonic as now # pylint: disable=no-name-in-module
+
+    # have to convert to a timedelta for rest of code to read
+    from future.backports.datetime import timedelta # pylint: disable=no-name-in-module
```

Comment:
Doesn't datetime include timedelta in python 3.3?

Reply:
datetime isnt imported at 3.3 thus why its imported from futures.

Reply:
https://docs.python.org/3/library/datetime.html

Doesn't seem to say it isn't supported at 3.3...